src/System.Net.Primitives/src/System/Net/SocketAddressPal.Windows.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Net.Sockets;
+using System.Text;
+
+namespace System.Net
+{
+    internal static class SocketAddressPal
+    {
+        public const int IPv6AddressSize = 28;
+        public const int IPv4AddressSize = 16;
+        public const int DataOffset = 2;
+
+        public static unsafe AddressFamily GetAddressFamily(byte[] buffer)
+        {
+            return (AddressFamily)BitConverter.ToInt16(buffer, 0);
+        }
+
+        public static unsafe void SetAddressFamily(byte[] buffer, AddressFamily family)
+        {
+#if BIGENDIAN
+            buffer[0] = unchecked((byte)((int)family >> 8));
+            buffer[1] = unchecked((byte)((int)family));
+#else
+            buffer[0] = unchecked((byte)((int)family));
+            buffer[1] = unchecked((byte)((int)family >> 8));
+#endif
```

Comment:
This isn't doing a host <-> network transformation; it's just writing the bytes out in host order. Ideally this would use BitConverter, but that type does not have any method that writes into an existing array.